driver-core/src/test/functional/com/mongodb/internal/connection/DefaultConnectionPoolTest.java

```diff
@@ -443,6 +443,41 @@ public void checkoutHandOverMechanism() throws InterruptedException, TimeoutExce
         }
     }
 
+    @Test
+    public void invalidateAfterCloseMustNotThrow() {
+        provider = new DefaultConnectionPool(
+                SERVER_ID,
+                connectionFactory,
+                ConnectionPoolSettings.builder().maxSize(1).build(),
+                mockSdamProvider());
+        provider.ready();
+        provider.close();
+        provider.invalidate(null);
+    }
+
+    @Test
+    public void invalidateConcurrentWithCloseMustNotThrow() throws ExecutionException, InterruptedException {
+        Future<?> backgroundTask = null;
+        for (int i = 0; i < 10_000; i++) {
```

Comment:
Does the number of iterations need to be quite so high to reliably generate a failure if there is a regression?  I was able to get it to fail (after commenting out the bug fix) with just one iteration.